lib/dtutils.lua

```diff
@@ -0,0 +1,866 @@
+  table.sort(result)
+  for _,img in pairs(result) do
+    images[cnt] = dtutils.get_basename(img)
+    cnt = cnt + 1
+  end
+
+  cnt = cnt - 1
+
+  if cnt > 1 then
+    if cnt > max_distinct_names then
+      -- take the first and last
+      outputFileName = images[1] .. name_separator .. images[cnt]
+    else
+      -- join them
+      outputFileName = dtutils.join(images, name_separator)
+    end
+  else
+    -- return the single name
+    outputFileName = images[cnt]
+  end
+
+  return outputFileName
+end
+
+--[[
+  NAME
+    dutils.chop_filetype - remove a filetype from a filename
+
+  SYNOPSIS
+    result = dtutils.chop_filetype(path)
+      path - a filename with or without a path
+
+  DESCRIPTION
+    chop_filetype removes the filetype from the filename
+
+  RETURN VALUE
+    result - the path and filename without the filetype
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.chop_filetype(path)
+  local length = dtutils.get_filetype(path):len() + 2
+  return string.sub(path, 1, -length)
+end
+
+--[[
+  NAME
+    dutils.prequire - a protected lua require
+
+  SYNOPSIS
+    result = dtutils.prequire(req_name)
+      req_name - name of thu lua code to load
+
+  DESCRIPTION
+    prequire is a protexted require that can survive an error in the code being loaded without
+    bringing down the calling routine.
+
+  RETURN VALUE
+    result - the code or true on success, otherwise an error message
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.prequire(req_name)
+  dt.print_error("Loading " .. req_name)
+  local status, lib = pcall(require, req_name)
+  if status then
+    dt.print_error("Loaded " .. req_name)
+  else
+    dt.print_error("Error loading " .. req_name)
+  end
+  log.msg(log.debug, "return type of lib is ", type(lib))
+  return lib
+end
+
+--[[
+  NAME
+    dutils.push - push a value on a stack
+
+  SYNOPSIS
+    result = dtutils.push(stack, value)
+      stack - a table being used as the stack
+      value - the value to be put on the stack
+
+  DESCRIPTION
+    Push a value on a stack
+
+  RETURN VALUE
+    result - false if stack isn't a table, true on success
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.push(stack, value)
+  success = false
+  if type(stack) == "table" then
+    table.insert(stack, value)
+    success = true
+  end
+end
+
+--[[
+  NAME
+    dutils.pop - pop a value from a stack
+
+  SYNOPSIS
+    result = dtutils.pop(stack)
+      stack - a table being used as a stack
+
+  DESCRIPTION
+    Remove the last value pushed on the stack and return it
+
+  RETURN VALUE
+    result = nil if stack isn't a table or the stack is empty, otherwise the value removed from the table
+
+  ERRORS
+    
+
+  
+]]
+
+function dtutils.pop(stack)
+  if type(stack) == "table" then
+    if #stack >= 1 then
+      return table.remove(stack)
+    else
+      return nil
+    end
+  else
+    return nil
+  end
+end
+
+--[[
+  NAME
+    dutils.fixSlliderFloat - correct decimal point problems with slidets and environments
```

Comment:
This came from official/enfuse.lua.  It was in the TODO list so I fixed it. I've not tried changing my locale to something that uses the comma as a decimal separator, so I don't know.  This is a probably a better question for @houz.